main.tf

```diff
@@ -46,9 +46,9 @@ resource "aws_efs_mount_target" "default" {
   file_system_id = join("", aws_efs_file_system.default.*.id)
   ip_address     = var.mount_target_ip_address
   subnet_id      = var.subnets[count.index]
-  security_groups = compact(
-    (concat(
-      [module.security_group.id],
+  security_groups = sort(
+    compact(concat(
+      var.create_security_group ? [module.security_group.id] : [],
```

Comment:
Not really necessary to test `var.create_security_group` since when it is false, `module.security_group.id` should be empty, but OK.

Please also update Security Group version on line 91 to 1.0.1.